test/ad/utils.jl

```diff
@@ -3,6 +3,7 @@ const AD = get(ENV, "AD", "All")
 
 function test_ad(f, x, broken=(); rtol=1e-6, atol=1e-6)
     finitediff = FiniteDifferences.grad(central_fdm(5, 1), f, x)[1]
+    et = eltype(finitediff)
```

Comment:
Why is this needed? Shouldn't Enzyme return the correct types automatically?

Reply:
In forward mode it returns tuples, and if the gradient is empty, the result is a `Tuple{}`. This resulted in comparing an empty `Float64[]` to an empty `Union{}[]`, which failed. See https://github.com/EnzymeAD/Enzyme.jl/issues/1584

Reply:
But the gradient should never be empty? Such a test would be quite useless, so I assume we don't run into this special case here? So maybe a simple `collect` (without specifying the element type) would be sufficient?

Reply:
I'd suggest not testing AD in the case `d = 1` - or just checking that the gradient is empty. We already handle this case in a special way in e.g. https://github.com/TuringLang/Bijectors.jl/blob/86544752e41a6a4aab0e1607fbf1e53a8c548a64/test/bijectors/corr.jl#L32-L33.

Reply:
It passes now though, and I don't really see a downside to testing it? Good to know for instance that nothing crashes even if you hit this corner case.

Reply:
No, I also would prefer to not remove the test completely (even though I think it's of very limited use)

> just checking that the gradient is empty

Reply:
I meant something differently - removing the `eltype`/`collect` completely and only add a special case to this weird test of the `CorrBijector` since we already have special cases for `d = 1` there anyway.

Reply:
Yeah, that's how I understood you, but that would require something like adding another argument to `test_ad` called `expect_empty` that would skip comparing to finitediff and instead check that the gradient has length 1 (for all AD backends) and setting that argument to `d == 1`, which to me seems more complicated, with a bunch of `if` statements, compared adding the one-liner enforcing `eltype`. I can do it if you prefer it, I just don't see the benefit.